pkg/auth/cli.go

```diff
@@ -48,6 +50,7 @@ func GetLoginFlags(flags *LoginOptions) *pflag.FlagSet {
 	fs.StringVarP(&flags.Username, "username", "u", "", "Username for registry")
 	fs.BoolVar(&flags.StdinPassword, "password-stdin", false, "Take the password from stdin")
 	fs.BoolVar(&flags.GetLoginSet, "get-login", false, "Return the current login user for the registry")
+	fs.BoolVar(&flags.AcceptRepositories, "accept-repositories", false, "Allow namespaces or repositories rather than just registries")
```

Comment:
Who is ever going to do this?  Shouldn't this be default and not an option?

Reply:
I think the migration strategy would be to introduce it as optional flag which can be later on turned on by default. What are your thoughts on that @vrothberg @mtrmac?

Reply:
This was discussed in this PR already around https://github.com/containers/common/pull/659#issuecomment-872999764 : At least one person did do this in https://github.com/containers/podman/issues/1690 .

---

I think any _new_ callers of c/common/pkg/auth, if any (i.e. new tools) should fairly clearly only use the new semantics (reject URLs and not ignore the within-registry paths).

As for the three current callers… just flipping this flag feels problematic given the known users, and that we would change the semantics of some inputs. Shouldn’t there be at least a period where the tools warn but still accept the old syntax?

--- 

Note: Even after we flip the flag to default on, we will have to silently accept it afterwards. That’s a bit of an argument in favor of not having the flag in the first place — I personally don’t think it’s a strong enough reason alone, though.

Reply:
I would rather have it support both formats and not blow up in the users face.  If at some time in the future Podman 4.0, you want to block the old format we could to it then.  No reason for humans to have to deal with this.

Reply:
We are changing the semantics of the `quay.io/foo:label` input as used by that user above. There’s no way for users to “not deal with this”: either we ask for an opt-in flag to use the new semantics (as this PR does), or we change the existing semantics and potentially break users.

[We could continue to silently support https://host.com/and/this/is/ignored indefinitely, though, at some risk of confusing users who might think the path means something.]